src/platform/sdl2-platform.c

```diff
@@ -273,8 +274,9 @@ static boolean pollBrogueEvent(rogueEvent *returnEvent, boolean textInput) {
             } else if (key == SDLK_F11 || key == SDLK_F12
                     || key == SDLK_RETURN && (SDL_GetModState() & KMOD_ALT)) {
                 // Toggle fullscreen
+                fullScreen = !fullScreen;
                 SDL_SetWindowFullscreen(Win,
-                    (SDL_GetWindowFlags(Win) & SDL_WINDOW_FULLSCREEN_DESKTOP) ? 0 : SDL_WINDOW_FULLSCREEN_DESKTOP);
+                    fullScreen ? SDL_WINDOW_FULLSCREEN_DESKTOP : 0);
```

Comment:
This causes problems when the user can change full-screen without using the Brogue logic, e.g. via window manager shortcuts on Linux. Could you revert it back?

Reply:
Sure, I can do that.  I may also rename the variable `initialFullScreen` or `fullScreenOnStart` in that case.

Reply:
I've left the patch as a separate commit in case you want to review it individually.  I'd prefer to fix it up into the `--full-screen` commits once it's ready to go.